src/Microsoft.Identity.Web.TokenAcquisition/MsalAspNetCoreHttpClientFactory.cs

```diff
@@ -17,7 +17,9 @@ public MsalAspNetCoreHttpClientFactory(IHttpClientFactory httpClientFactory)
 
         public HttpClient GetHttpClient()
         {
-            HttpClient httpClient = _httpClientFactory.CreateClient();
+            // The HttpClient will be created with the name of the client registered in the DI container
+            // If the named client(dSTS) is not found, the factory will fall back to the default client
+            HttpClient httpClient = _httpClientFactory.CreateClient("dSTS");
```

Comment:
Id Web is used by services outside of Microsoft, this will not work.

Reply:
I think that the design we had agreed on is, in the TokenAcquisition constructor:

https://github.com/AzureAD/microsoft-identity-web/blob/77df2a63d759beb5bbeb9c1260c8efdffd9fff94/src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs#L101

use the locator pattern to get a IMsalAspNetCoreHttpClientFactory if there is one in the service provider.

 _httpClientFactory = serviceProvider.GetService<IMsalAspNetCoreHttpClientFactory> ?? new MsalAspNetCoreHttpClientFactory(httpClientFactory);